bundles/http/Citrix/SAPHana/script.js

```diff
@@ -0,0 +1,913 @@
+        body: JSON.stringify(body)
+    });
+
+    console.log('## [respPOST] -> ' + JSON.stringify(body))
+
+    // AFTER POST
+    if (!respPOST.ok) {
+        throw new Error(`Could not create time entry -> Status: ${respPOST.status} / Response: ${JSON.stringify(respPOST)})`);
+    }
+
+    let postJson = await respPOST.json()
+
+    const requestPersonWorkAgreementExternalId = postJson.d.PersonWorkAgreementExternalID
+    const requestCompanyCode = postJson.d.CompanyCode
+    const requestTimeSheetRecord = postJson.d.TimeSheetRecord
+
+    const timeEntryUrl = `API_MANAGE_WORKFORCE_TIMESHEET/TimeSheetEntryCollection(PersonWorkAgreementExternalID='${requestPersonWorkAgreementExternalId}',CompanyCode='${requestCompanyCode}',TimeSheetRecord='${requestTimeSheetRecord}')`
+
+    const getCreatedTimeEntry = await client.fetch(timeEntryUrl, defaultGetOptions);
+
+    if (!getCreatedTimeEntry.ok) {
+        throw new Error(`Could not retrieve data from time entry (${getCreatedTimeEntry.status}: ${getCreatedTimeEntry.statusText})`);
+    }
+
+    const timeEntryResponse = await getCreatedTimeEntry.json()
+
+    const TimeSheetDateTimeStamp =  timeEntryResponse.d.TimeSheetDate.replace(/[^.\d]/g, '');
+
+    timeEntries.push({
+        id: parseInt(timeEntryResponse.d.TimeSheetRecord),
+        ts_id: timeEntryResponse.d.TimeSheetRecord,
+        ts_date: new Date(parseInt(TimeSheetDateTimeStamp)),
+        company_code: timeEntryResponse.d.CompanyCode,
+        person_work_agreement: timeEntryResponse.d.PersonWorkAgreement,
+        person_work_agreement_external_id: timeEntryResponse.d.PersonWorkAgreementExternalID,
+        ts_df_controlling_area: timeEntryResponse.d.TimeSheetDataFields.ControllingArea,
+        ts_df_hours_unit_of_measure: timeEntryResponse.d.TimeSheetDataFields.HoursUnitOfMeasure,
+        ts_df_purchase_order_item: timeEntryResponse.d.TimeSheetDataFields.PurchaseOrderItem,
+        ts_df_receiver_cost_center: timeEntryResponse.d.TimeSheetDataFields.ReceiverCostCenter,
+        ts_df_recorded_hours: timeEntryResponse.d.TimeSheetDataFields.RecordedHours,
+        ts_df_recorded_quantity: timeEntryResponse.d.TimeSheetDataFields.RecordedQuantity,
+        ts_df_ts_overtime_category: timeEntryResponse.d.TimeSheetDataFields.TimeSheetOvertimeCategory,
+        ts_df_ts_task_component: timeEntryResponse.d.TimeSheetDataFields.TimeSheetTaskComponent,
+        ts_df_ts_task_level: timeEntryResponse.d.TimeSheetDataFields.TimeSheetTaskLevel,
+        ts_df_ts_task_type: timeEntryResponse.d.TimeSheetDataFields.TimeSheetTaskType,
+        ts_df_ts_note: timeEntryResponse.d.TimeSheetDataFields.TimeSheetNote,
+        ts_operation: timeEntryResponse.d.TimeSheetOperation,
+        ts_status: timeEntryResponse.d.TimeSheetStatus,
+        user_email: user_email
+    })
+
+    dataStore.save("TimeEntries", timeEntries);
+    dataStore.save("TimeEntriesClock", timeEntries);
+}
+
+async function clockManager({dataStore, client, actionParameters}) {
```

Comment:
A lot of the code is repeated from the `createTimeEntry`. Maybe create one function with parameters, or put the common code in separated functions and reuse?

Reply:
Done